tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -780,6 +780,51 @@ public:
    void AcceptVisitor(Detail::RFieldVisitor &visitor) const override;
 };
 
+/// The field for a class in unsplit mode, which is using ROOT standard streaming
+class RUnsplitField final : public RFieldBase {
+private:
+   class RUnsplitDeleter : public RDeleter {
+   private:
+      TClass *fClass;
+
+   public:
+      explicit RUnsplitDeleter(TClass *cl) : fClass(cl) {}
+      void operator()(void *objPtr, bool dtorOnly) final;
+   };
+
+   TClass *fClass = nullptr;
+   ClusterSize_t fIndex; ///< number of bytes written in the current cluster
+
+private:
+   RUnsplitField(std::string_view fieldName, std::string_view className, TClass *classp);
```

Comment:
Can `className` be a different spelling than `classp->GetName()` and/or can `classp` be a `nullptr`?

Reply:
Yes, both is possible. I added a comment and a test.